src/libcore/iter/adapters/mod.rs

```diff
@@ -1158,6 +1184,45 @@ impl<I: Iterator> Iterator for Peekable<I> {
     }
 }
 
+#[stable(feature = "double_ended_peek_iterator", since = "1.37.0")]
+impl<I> DoubleEndedIterator for Peekable<I> where I: DoubleEndedIterator {
+    #[inline]
+    fn next_back(&mut self) -> Option<Self::Item> {
+        self.iter.next_back().or_else(|| self.peeked.take().and_then(|x| x))
```

Comment:
If `Peekable` is going to be DEI, then should it have a `peek_back()` too?  But that has space cost, which might be why it wasn't already double-ended?

Reply:
Yeah, that may be why this implementation was missing, but I don't think it needs to have a `peek_back()` method for it to be useful as a DEI. If anyone needs to be able to peek from both sides, a separate type would probably make more sense because of the memory implications you mentioned.

The only surprising thing about it being a DEI would perhaps be that `nth_back` can't be implemented efficiently.